apply_DDT.py

```diff
@@ -0,0 +1,562 @@
+    '''
+    plt.tight_layout()
+    plt.savefig(f'plots/{plt_name}.png')  
+    plt.savefig(f'plots/{plt_name}.pdf')  
+
+def bdt_ddt_inputs(col, lumi, all_features):
+
+    # Storing column features
+    X = []
+    weight = []
+
+    # Apply the signal region
+    if isinstance(col, list):
+        col = [apply_rt_signalregion(c) for c in col]
+    else:
+        col = apply_rt_signalregion(col)
+
+    # Test if more than one column
+    if isinstance(col,list):
+        for icol in col:  
+            # Grab the input features and weights
+            X.append(icol.to_numpy(all_features))
+            weight.append(icol.xs / icol.cutflow['raw'] * lumi * icol.arrays['puweight'])
+        X = np.concatenate(X)
+        weight = np.concatenate(weight)
+
+    # If only one column
+    else: 
+        X = col.to_numpy(all_features)
+        weight = col.xs / col.cutflow['raw'] * lumi * col.arrays['puweight']
+  
+    # grab rt
+    rt = X[:,-1]
+    X = X[:,:-1] # remove it from X so that eventually it can be used for BDT scores
+  
+    # grab rho
+    rho = X[:,-1]
+    X = X[:,:-1] # remove it from X so that eventually it can be used for BDT scores
+  
+    # grab mT
+    mT = X[:,-1]
+    X = X[:,:-1] # remove it from X so that eventually it can be used for BDT scores
+  
+    # grab pT
+    pT = X[:,-1]
+    X = X[:,:-1] # remove it from X so that eventually it can be used for BDT scores
+
+    return X, pT, mT, rho, weight
+
+
+#------------------------------------------------------------------------------
+# The Main Function -----------------------------------------------------------
+#------------------------------------------------------------------------------
+
+def main():
+    
+    # Parse arguments and take the results
+    args = parse_arguments()
+
+    bkg_files = args.bkg_files
+    sig_files = args.sig_files
+    model_file = args.bdt_file
+    ddt_map_file = args.ddt_map_file
+    lumi = args.lumi
+    sig_bdt_cut = args.sig_bdt_cut
+    bdt_cuts = args.bdt_cuts
+    plots = args.plot
+    verbosity = args.verbosity
+
+    set_mpl_fontsize(18, 22, 26)
+
+    # grab training features for specific input model
+    global training_features
+    training_features = read_training_features(model_file)
+
+    # add ddt necessary variables in addition to the bdt input features
+    all_features = training_features + ['pt', 'mt', 'rho', 'rt'] # rho is an important variable for applying the decorrelation
+
+    # Grab the bkg data
+    bkg_cols = [Columns.load(f) for f in glob.glob(bkg_files)]
+    X, pT, mT, rho, bkg_weight = bdt_ddt_inputs(bkg_cols, lumi, all_features)
+
+    # _____________________________________________
+    # Open the trained models and get the scores
+
+    bkg_score = {}
+
+    xgb_model = xgb.XGBClassifier()
+    xgb_model.load_model(model_file)
+    with time_and_log(f'Calculating xgboost scores for {model_file}...'):
+        bkg_score = xgb_model.predict_proba(X)[:,1]
```

Comment:
this pattern repeats 5 times in the code (including the existing instance in `common.py`). define a function for this to reduce duplication

Reply:
Done, needs testing.